includes/User.php

```diff
@@ -2551,6 +2554,22 @@ public function setGlobalPreference($preference, $value) {
 	}
 
 	/**
+	 * @param array $preferences preferenceName->Value
+	 */
+	public function setGlobalPreferences( $preferences ) {
+		global $wgPreferencesUseService;
+		if ($wgPreferencesUseService) {
+			$this->sanitizePropertyArray($preferences);
+			$this->userPreferences()->setPreferences($this->mId, $preferences);
```

Comment:
I guess the naming of the function `setPreferences` is misleading. It doesn't actually make a service call to set a user's preferences. It's probably better named as `loadPreferencesFromCache`, since that's what it's used for in [User.php](https://github.com/Wikia/app/blob/dev/includes/User.php#L363). 

Can we change the name of the `UserPreferences->setPreferences` function and introduce a new function called `setMultiple` that takes the `$userId` and `$preferences` array as input?